### DISCUSSION TITLE
NOTHING HELPS - Timed out fetching a new connection from the connection pool.

### DISCUSSION THREAD

**Discussion by idbd1** [prisma/prisma#24170] [Q&A] [answered]
### Question

Timed out fetching a new connection from the connection pool. (...) (Current connection pool timeout: 10, connection limit: 49).

Hi, I've been having this issue when fetching and I'm running out of ideas how to tackle the problem. The request shows up as pending for about 15-50 seconds every time I make the request, then it ends up showing the correct data.

Here's what I tried so far:
1. I minimised number of new started prisma connections to two.
2. I had increased connection limit to 100, but it did not resolve the issue, so I reverted it back to the default.
3. I removed a lot of unnecessary data that was fetched from the request.

Any other ideas what I could try?

Many thanks.

### Information about Prisma Schema, Client Queries and Environment (optional)

I'm using node v18.18.2, prisma is on v5.13.0

**Comment by ludralph** [comment]
Hi @dominikabieder 👋 

Thank you for raising this question.

Can you increase the `pool_timeout` to a higher value? Connection creation plus query time may sometimes take longer than the currently configured timeout. Do let us know what happens when you do this.

**Reply by idbd1** [reply, marked as answer]
thanks, I tried it but it didn't help.

Turns out I was using a recursive function that made a very high number of requests, which in turn opened many many  connections. I had to refactor the request to get rid of the problem.
